docs/rules/number-literal-case.md

```diff
@@ -0,0 +1,27 @@
+# Enforce a lowercase literal identifier and an uppercase value
+
+Enforces a convention in defining number literals where the literal identifier is written in lowercase and the value in uppercase.
+
```

Comment:
We should include a reasoning here. Maybe:

> Differentiating the casing of the identifier and value clearly separates them and makes your code more readable.

Improvement suggestions welcome.